go-controller/pkg/clustermanager/userdefinednetwork/notifier/namespace.go

```diff
@@ -0,0 +1,64 @@
+package notifier
+
+import (
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+	corev1informer "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/controller"
+)
+
+type NamespaceReconciler interface {
+	ReconcileNamespace(key string) error
+}
+
+// NamespaceNotifier watches Namespaces objects and notify subscribers upon change.
+// It enqueues the reconciled object keys in the subscribing controllers workqueue.
+type NamespaceNotifier struct {
+	Controller controller.Controller
+
+	subscribers []NamespaceReconciler
+}
+
+func NewNamespaceNotifier(nsInformer corev1informer.NamespaceInformer, subscribers ...NamespaceReconciler) *NamespaceNotifier {
```

Comment:
we know there will only be 1 user of this notifier ever, right?
If so, can we just merge it into the main controller, then it will be much easier to link the namespace controller to the real reconcile function without extra interfaces in between?

Reply:
I followed the same pattern as with the NAD notifier, because it seem that having them in separate package keep the boilerplate out of the controller business logic in controller.go, and reduce so much noise.
WDYT?